### ISSUE TITLE
[Bug]: no-deprecated does not report deprecated object literal properties

### ISSUE THREAD

**Issue by Norbiros** [oxc-project/tsgolint#943]
## Description

`typescript/no-deprecated` does not report deprecated properties when they are used as keys in contextually typed object literals.

I noticed this while testing Nuxt's deprecated `statusCode` property in the `showError` utility.

## Reproduction

1. Create a new file `file.ts` with this content:
```
interface ErrorOptions {
  /** @deprecated Use status instead. */
  statusCode?: number

  status?: number
}

declare function showError(error: ErrorOptions): void

showError({
  statusCode: 500,
})
```

2. Execute oxlint with `--type-aware`
```bash
$ pnpm oxlint --type-aware --deny typescript/no-deprecated file.ts 
Found 0 warnings and 0 errors.
Finished in 221ms on 1 file with 111 rules using 16 threads.
```

We clearly use deprecated symbol, but no issue is being raised.

**Comment by camc314** [comment]
typescript-eslint doesn't appear to report this case https://github.com/typescript-eslint/typescript-eslint/issues/10643

**Comment by camc314** [comment]
I've got a PR to fix this, but I'm holding merging it for a bit just because it might cause really annoying behaivour.

For example if a property is required on an object, but deprecated, you are going to either have to `//ts-expect-error` or `// oxlint-disable-next-line` to silence it - keeping the prop will cause a linter error, but removing it will cause a TS error

**Comment by Norbiros** [comment]
It might just be me, but I’d prefer to get a clear warning when I’m using a deprecated property. It would motivate me to contact the library authors or resolve the issue myself. IMHO, it’s better to explicitly acknowledge the use of a deprecated & required property and add an `oxlint-disable` comment (or `@ts-expect-error` in some edge cases) than to silently ignore it.

I don't think there is a possible solution to this problem. You always might encounter conflict like this (ts vs jsdoc), so it would be better to introduce it earlier, than in random minor release.

**Comment by camc314** [comment]
>  so it would be better to introduce it earlier, than in random minor release.
fixing it now vs in future would just be fixing it in different minor releases

**Comment by camc314** [comment]
cc @bradzacher should we ship a fix for this? what do you think?

**Comment by bradzacher** [comment]
Yeah fix it - the issue is open for someone to add the feature to typescript-eslint its just that nobody has been motivated to action it.

I'd agree that reporting even on required props would be fine. If a library is doing something dumb like that then people would be right to complain to them to fix it - deprecating a required property makes little logical sense.
